### PR TITLE
[16.0][MIG] sale_loyalty_delivery: Migration to version 16.0

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT44345

@pedrobaeza @chienandalu please review

### CODE REVIEW

File: openupgrade_scripts/scripts/sale_loyalty_delivery/16.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        ("reward_type", None, None),
+    ],
+}
+
+
+def check_for_free_shipping_promotions(env):
+    """
+    Since the values of the selection reward_type field only change in the
+    sale_loyalty_delivery module, a copy of the column will only be made in case there
+    is a free delivery promotion. In case of such a promotion, a copy of the column will
```
Comment (chienandalu) [inline openupgrade_scripts/scripts/sale_loyalty_delivery/16.0.1.0/pre-migration.py:16]: Shouldn't we copy the column in the `coupon` `pre-mig`? Isn't it very likely that at this point we won't have this info?

Comment (pilarvargas-tecnativa) [reply openupgrade_scripts/scripts/sale_loyalty_delivery/16.0.1.0/pre-migration.py:16]: This field comes as it is from v15, only the selection value is modified in this module so there is no loss of previous information :)

**Review by chienandalu — APPROVED** [review]
Ok :+1:

**Comment by pedrobaeza** [comment]
/ocabot migration sale_loyalty_delivery

<dependency>Depends on : 

- loyalty_delivery : TODO</dependency>

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**